src/OrleansTestKit/Streams/TestStream.cs

```diff
@@ -91,8 +91,17 @@ public Task OnNextAsync(T item, StreamSequenceToken token = null)
         public Task OnErrorAsync(Exception ex) =>
             Task.WhenAll(_observers.ToList().Select(o => o.OnErrorAsync(ex)));
 
-        public Task OnNextBatchAsync(IEnumerable<T> batch, StreamSequenceToken token = null) =>
-            throw new NotImplementedException();
+        public async Task OnNextBatchAsync(IEnumerable<T> batch, StreamSequenceToken token = null)
+        {
+            if (batch == null)
+            {
+                throw new ArgumentNullException(nameof(batch));
+            }
+            foreach (var item in batch)
+            {
+                await OnNextAsync(item, token).ConfigureAwait(false);
```

Comment:
If an error occurs processing an item, the for loop will terminate. Is that desired/expected?

Reply:
I added handling / rethrowing an AggregateException to allow all messages to be sent, but still bubble up any exceptions.

Reply:
That sounds like a nice approach. Is that the same behavior as Orleans?

Reply:
It depends on which stream provider you're using.  We use SQS, for instance.  Since the message is pushed off to an out-of-process message queue, an exception from the subscriber wouldn't bubble up to the publisher.  Simple Message Stream Provider, on the other hand, would propagate the exception back to the caller for OnNextAsync.  SMS doesn't currently support OnNextBatchAsync https://github.com/dotnet/orleans/blob/10af0f4af588cd4aa45cb3e250dfbffa389d59c7/src/Orleans.Streaming/SimpleMessageStream/SimpleMessageStreamProducer.cs#L121.